homeassistant/components/abode/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for the Abode Security System component."""
+import logging
+import voluptuous as vol
+from abodepy import Abode
+from abodepy.exceptions import AbodeAuthenticationException
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AbodeFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Abode."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self.data_schema = {}
+        self.data_schema[vol.Required(CONF_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if not user_input:
+            return await self._show_form()
+
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+
+        try:
+            await self.hass.async_add_executor_job(Abode, username, password, True)
+
+        except AbodeAuthenticationException as error:
+            if error.errcode == 400:
+                return await self._show_form({"base": "invalid_credentials"})
+            return await self._show_form({"base": "abode_error"})
+
+        return self.async_create_entry(
+            title=user_input[CONF_USERNAME],
+            data={CONF_USERNAME: username, CONF_PASSWORD: password},
+        )
+
+    async def _show_form(self, errors=None):
```

Comment:
We don't use `await` inside so this doesn't need to be a coroutine. Decorate it with `@callback` instead.